config/prow/cluster/deck_deployment.yaml

```diff
@@ -64,6 +64,13 @@ spec:
         # Use KUBECONFIG envvar rather than --kubeconfig flag in order to provide multiple configs to merge.
         - name: KUBECONFIG
           value: "/etc/kubeconfig/config:/etc/kubeconfig-build-test-infra-trusted/kubeconfig:/etc/kubeconfig-build-k8s-prow-builds/kubeconfig:/etc/kubeconfig-build-rules-k8s/kubeconfig"
+        # AWS_ variables needed to assume role to access the prow-build-cluster EKS cluster.
+        - name: AWS_ROLE_ARN
+          value: arn:aws:iam::468814281478:role/Prow-EKS-Admin
```

Comment:
Not for me to judge what is acceptable here, but seems like this could be locked down a bit.

Reply:
If you mean not sharing it publicly, that would be more complicated and we also share ARNs for other stuff for some other jobs and components, so it shouldn't be an issue.

Reply:
I'd have to dig through the code to see what permissions Prow-EKS-Admin has, but it just seems like deck, hook, etc might not need full root typer permissions. If that has been working in the past, it probably isn't an issue though.

Reply:
The tokens used by Prow to access build clusters have always been a service account bound to the `cluster-admin` role(I agree it is overprivileged).

I have configured it to do the same thing, that role will have cluster-admin permissions.

https://docs.prow.k8s.io/docs/build-clusters/
https://docs.prow.k8s.io/docs/getting-started-deploy/#run-test-pods-in-different-clusters